### PR TITLE
Edit

### PR DESCRIPTION
**Pull request by maxmurdoch**
Everything that isn’t a correction is just a proposal. Accept or reject
at will!

@hmarr and @harrison please review <3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,49 +43,50 @@ Reference: http://www.mnot.net/blog/2012/04/13/json_or_xml_just_decide
 ## General guidelines
 - A URL identifies a resource.
 - URLs should include nouns, not verbs.
-- Use plural nouns only for consistency (no singular nouns).
-- Use HTTP verbs (GET, POST, PUT, DELETE) to operate on the collections and elements.
-- Never nest resources - it enforces relationships that could change, and makes clients harder to write.
-  - Use filtering: `/payments?subscription=xyz` rather than `/subscriptions/xyz/payments`
-- API versions should be represented as dates documented in a changelog. Version number should not be in the url.
+- For consistency, only use plural nouns (e.g. "posts" instead of "post").
+- Use HTTP verbs (GET, POST, PUT, DELETE) to operate on collections and elements.
```
Comment (maxmurdoch) [inline README.md]: `single resources or collections`?

File: README.md [hunk 2]
```diff
@@ -139,32 +138,40 @@ Values in keys:
...
 Validation and resource errors are nested in the top level error under `errors`.
 
 The error is nested in `error` to make it possible to add, for example, deprecation errors on successful requests.
 
-The HTTP status `code` is used as a top level error, `type` is used a sub error. Nested `errors` may have more specific `type` errors like `invalid_field`.
+The HTTP status `code` is used as a top level error, `type` is used as a sub error, and nested
+`errors` may have more specific `type` errors, such as `invalid_field`.
+
+### Formating errors vs integration errors
+Formatting errors should be separate from errors the integration should handle.
```
Comment (maxmurdoch) [inline README.md]: double use of should is clumsy

File: README.md [hunk 3]
```diff
@@ -139,32 +138,40 @@ Values in keys:
...
+Always return string ids. Some languages, like JavaScript, don't support big ints. Serialize/deserialize ints to strings if storing ids as ints.
 
 ## Error handling
-Error responses should include a message for the user, internal error type (corresponding to some specific internally determined constant, which must be represented as a string), links where developers can find more info.
+Error responses should include a message for the user, an internal error type (corresponding to some
+specific internally determined constant represented as a string), and links to info for developers.
 
-There must only be one top level error. Errors should be returned in turn. This makes internal error logic simpler. It also makes it easier for API consumers to deal with errors.
+There must only be one top level error. Errors should be returned in turn. This makes internal error
+logic and dealing with errors as a consumer of the API easier.
```
Comment (maxmurdoch) [inline README.md]: this is emacs line wrapping. I don't know if it'll produce ugly markdown output. If it does I'll fix it in a follow up.

File: README.md [hunk 4]
```diff
@@ -139,32 +138,40 @@ Values in keys:
...
 
-- Top level errors MUST implement `request_id`, `type`, `reason`, `code`,`message`.
-- `type` MUST relate to the `reason`, use it to categorise the error, e.g. `api_error`.
+- Top level errors MUST implement `request_id`, `type`, `reason`, `code` and `message`.
+- `type` MUST relate to the `reason`. For example, use it to categorise the error: `reason: api_error`.
 - `reason` MUST be specific to the error.
 - `message` MUST be specific.
-- Top level errors MAY implement `documentation_url`, `request_url `, `id `.
-- Only return `id` for server errors (5xx). The id should point to the exception you track internally.
+- Top level errors MAY implement `documentation_url`, `request_url `and `id`.
```
Comment (maxmurdoch) [inline README.md]: spacing

File: README.md [hunk 5]
```diff
@@ -319,16 +337,14 @@ Paginated results are always enveloped:
     "limit": 50
   },
   "payments": [{
-    ...
-  }]
+    …
+  },
+  …]
```
Comment (maxmurdoch) [inline README.md]: `alt + ;` is the mac shortcut for the ellipsis character

File: README.md [hunk 6]
```diff
@@ -417,11 +437,15 @@ Exceeding rate limit:
 ```
 
 ## CORS
-Support Cross Origin Resource Sharing (CORS) for AJAX requests. You can read the CORS W3C working draft, or this intro from the HTML 5 Security Guide.
+Support Cross Origin Resource Sharing (CORS) for AJAX requests.
+
+Resources:
+- [CORS W3C working draft](https://www.w3.org/TR/cors/)
+- [HTML5 Rocks](http://www.html5rocks.com/en/tutorials/cors/).
```
Comment (maxmurdoch) [inline README.md]: couldn't find the HTML 5 Sec guide link, so tentatively replaced it with this link. Could remove it from the list though. Or omit a resources list at all.

File: README.md [hunk 7]
```diff
@@ -266,36 +278,41 @@ Reference: https://stripe.com/docs/upgrades
...
+```http
 GET /refunds?payments=ID1,ID2&customer=ID1
 ```
 
 ## Pagination
 
 All list/index endpoints must be paginated by default.
-Pagination must be reverse chronological. To enable cursor based pagination, ids should be increasing.
+
+Pagination must be reverse chronological. Ids should increase to enable cursor based pagination.
```
Comment (harrison) [inline README.md]: could :hocho: ` Ids should increase to enable cursor based pagination.`, related to implementation and only one way of doing it

**Comment by harrison** [comment]
Awesome! :+1:

**Pull request merged**